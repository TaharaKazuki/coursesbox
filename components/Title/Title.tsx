import { FC, ReactNode } from 'react'
import styled from '@emotion/styled'
import { AppTheme } from '@/styles/themes'
import { boxShadow, borderRadius } from '@/components/styles'

const Section = styled.section<Props>`
  ${borderRadius};
  padding: 1vmin 4vmin 4vmin;
  background: ${({ theme }) => theme.components.background};
  color: ${({ theme }) => theme.font.regular};
  ${({ theme }) => boxShadow(theme.components.shadow1, theme.components.shadow2)}
`

type Props = {
  children?: ReactNode
  header?: string
  theme?: AppTheme
}

export const Title: FC<Props> = ({ header, children }) => (
  <Section>
    <h2>{header}</h2>
    {children}
  </Section>
)
