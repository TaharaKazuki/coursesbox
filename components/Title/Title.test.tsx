import React from 'react'

import { render } from '@/test-utils'

import { Title } from './Title'

describe('Title test case', () => {
  it('Title render check', () => {
    const { asFragment } = render(
      <Title header="Lorem ipsum dolor sit amet">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
        laboris
      </Title>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
