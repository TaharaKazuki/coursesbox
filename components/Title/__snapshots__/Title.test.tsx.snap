// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title test case Title render check 1`] = `
<DocumentFragment>
  <section
    class="css-1ahij4z"
  >
    <h2>
      Lorem ipsum dolor sit amet
    </h2>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
  </section>
</DocumentFragment>
`;
